src/compact/compact.js

```diff
@@ -0,0 +1,40 @@
+import { forEach } from "../for-each/for-each"
+import { is } from "../is/is"
+import { map } from "../map/map"
+import { filter } from "../filter/filter"
+import { curry } from "../curry/curry"
+
+const _compactObject = source => {
```

Comment:
Would skip here adding one more function call, `forEach`, and use a simple `for`:

```js
const _compactObject = source => {
  const result = { ...source }
  const keys = Object.keys(result)

  for (const key of keys) {
    if (!is(result[key])) {
      delete result[key]
    }
  }

  return result
}
```